### ISSUE TITLE
Invalid Camel Case Property Name Resolution for CamelCaseNamingStrategy when using CamelCasePropertyNamesContractResolver

### ISSUE THREAD

**Issue by zeshan-iqbal** [JamesNK/Newtonsoft.Json#1330]
### Source/destination types

```csharp
	public class Citizen
	{
		public bool IsUSCitizen {get; set;}
	}
```

### Source/destination JSON

```javascript
{"isusCitizen":false}
```

### Expected behavior

It should serialise "IsUSCitizen" property name to "isUSCitizen" instead of "isusCitizen".

### Actual behavior

It serialise "IsUSCitizen" property name to "isusCitizen" instead of "isUSCitizen".

### Steps to reproduce

```csharp

using System;
using Newtonsoft.Json;
using Newtonsoft.Json.Serialization;

					
public class Program
{
	
	public class Citizen
	{
		public bool IsUSCitizen {get; set;}
	}
	
	public void Main()
    {
		Citizen citizen = new Citizen();
		 
		string json = JsonConvert.SerializeObject(citizen, new JsonSerializerSettings 
    { 
        ContractResolver = new CamelCasePropertyNamesContractResolver() 
    });
		Console.WriteLine(json);
    }
}
```

As of my understanding CamelCasePropertyNamesContractResolver uses CamelCaseNamingStrategy to resolve property name and CamelCaseNamingStrategy resolve property name by calling StringUtils.ToCamelCase(name). Even if I'm missing something "isusCitizen" is not correct serialisation of "isUSCitizen".

**Comment by TylerBrinkley** [comment]
This was fixed in 8.0.3, see the last line of the [changelist](https://github.com/JamesNK/Newtonsoft.Json/releases/tag/8.0.3).

**Comment by zeshan-iqbal** [comment]
Thanks @TylerBrinkley. Based on your comment closing this issue.
